### ISSUE TITLE
Release new react-json-tree?

### ISSUE THREAD

**Issue by akx** [reduxjs/redux-devtools#490]
Hey there!

There are downstream projects (the one I have my eye on is [`@nteract/outputs`](https://github.com/nteract/outputs/blob/8ca441567e4f92ac32088caf01c737bfb41403c3/packages/outputs/package.json#L16) that could use a version of `react-json-tree` that doesn't have a dependency on `babel-runtime`.

Would it be possible to release a new version of `react-json-tree`?

**EDIT 20200721:** updated nteract/outputs link

**Comment by DDKnoll** [comment]
Can we please do this? We've had to clone the repo because you can't run babel 7 with this project because it specifies babel 6 as a dependency.  We just need somebody to publish the code thats in the repo

**Comment by Methuselah96** [comment]
@timdorr What is the plan for publishing? It seems like we should publish:

- `react-json-tree`: `0.12.0` (resolves https://github.com/reduxjs/redux-devtools/issues/480, https://github.com/reduxjs/redux-devtools/issues/490 (and https://github.com/reduxjs/redux-devtools/issues/487 if we get https://github.com/reduxjs/redux-devtools/pull/513 merged))
- `redux-devtools-inspector`: `0.12.0` (with `react-json-tree` at `0.12.0`) (resolves https://github.com/reduxjs/redux-devtools/issues/481, https://github.com/reduxjs/redux-devtools/issues/497)
- `redux-devtools`: `3.6.0` (resolves https://github.com/reduxjs/redux-devtools/issues/473)

**Comment by timdorr** [comment]
No specific plan. I haven't yet looked through the state of the code right now to know if they should be published and at what version (patch vs minor vs major). 

Maybe we can do a quick RC release so they don't show up on the `latest` npm tag and and break things for folks without some testing first. Plus, that will make sure I know how to publish these libraries (in case there are different steps than I'm used to)

**Comment by Methuselah96** [comment]
Makes sense. I've compiled an exhaustive changelog. Hopefully that will help with the publish. I've included a short summary under each change where appropriate.

Overall, it looks like almost all of the changes were internal changes (https://github.com/reduxjs/redux-devtools/pull/449 looks like just linting). Before we started, I don't think there had been any unpublished bug fixes or feature changes. `redux-devtools-inspector` is the only package that looks like it has changes that would cause a regression.

### Changelog

#### react-json-tree

 - Last published from this repo
 - Changes since last publish:
   -  Bump webpack-dev-server from 2.11.5 to 3.10.3 in /packages/react-json-tree/examples (https://github.com/reduxjs/redux-devtools/pull/512)
      - Not a dependency upgrade for library, just examples
   - [react-json-tree] Mark deprecated React lifecycle methods as unsafe (https://github.com/reduxjs/redux-devtools/pull/483)
   - Fix and update packages (https://github.com/reduxjs/redux-devtools/pull/449)
     - Drop `babel-runtime` as dependency
     - Lint changes
 - Changes that might have been missed from source repo:
   - None

#### redux-devtools-instrument

 - Last published from this repo
 - Changes since last publish:
   - Prettify (https://github.com/reduxjs/redux-devtools/commit/d3fcbb85d9f023042fea4e25b4374768173254f2)
     - Lint changes
   - Fix and update packages (https://github.com/reduxjs/redux-devtools/pull/449)
     - Lint changes
 - Changes that might have been missed from source repo:
   - None

#### redux-devtools

 - Last published from this repo
 - Changes since last publish:
   - Update react-redux peer dependency (https://github.com/reduxjs/redux-devtools/pull/482)
   - Fix broken link in redux-devtools README (https://github.com/reduxjs/redux-devtools/pull/477)
   - Fix and update packages (https://github.com/reduxjs/redux-devtools/pull/449)
     - Lint changes
 - Changes that might have been missed from source repo:
   - None

#### redux-devtools-log-monitor

 - Last published from https://github.com/gaearon/redux-devtools-log-monitor
 - Changes since last publish:
   - Fix and update packages (https://github.com/reduxjs/redux-devtools/pull/449)
     - Lint changes
   - Merge redux-devtools-log-monitor (https://github.com/reduxjs/redux-devtools/pull/432) (reconstructed diff: https://github.com/Methuselah96/redux-devtools/commit/aa8f08fc2bc7f66cbbece15d9e1b2f46db720df6)
     - No changes
 - Changes that might have been missed from source repo:
   - None

#### redux-devtools-core

 - Last published from this repo
 - Changes since last publish:
   - Upgrade jsondiffpatch (https://github.com/reduxjs/redux-devtools/pull/507)
   - Fix and update packages (https://github.com/reduxjs/redux-devtools/pull/449)
     - Lint changes
   - Move remotedev-app and remotedev-server (https://github.com/reduxjs/redux-devtools/pull/445)
 - Changes that might have been missed from source repo:
   - None

#### redux-slider-monitor

 - Last published from https://github.com/calesce/redux-slider-monitor
 - Changes since last publish:
   - Fix and update packages (https://github.com/reduxjs/redux-devtools/pull/449)
     - Lint changes
   - Move remotedev-app and remotedev-server (https://github.com/reduxjs/redux-devtools/pull/445)
     - Add missing `prepare` npm script for internal use
   - Add redux-slider-monitor (https://github.com/reduxjs/redux-devtools/pull/434) (reconstructed diff: https://github.com/Methuselah96/redux-devtools/commit/2b4dd808fad864d8e9cf61b11789c35361f373a3)
     - Lint changes
 - Changes that might have been missed from source repo:
   - None

#### redux-devtools-inspector

 - Last published from https://github.com/alexkuz/redux-devtools-inspector
 - Changes since last publish:
   - Difficult to list since the latest published version is from https://github.com/alexkuz/redux-devtools-inspector and the version merged into this repo is https://github.com/zalmoxisus/remotedev-inspector-monitor. They diverge at 0.9.4. It's probably worth spending some time looking into this and reconciling the differences.
 - Changes that might have been missed from source repo:
   - Noted in https://github.com/reduxjs/redux-devtools/pull/429:
     - I'm moving the fork maintained in remotedev-inspector-monitor and renaming it back to redux-devtools-inspector. It contains some different changes started from 0.9.4. Except adding flow, the upstream has 2 other important commits:
       - perf improvement, but it'll need some changes now because componentWillReceiveProps is getting deprecated;
       - adding maps/sets previews, but it conflicts with our custom types introduced before.

#### devui

 - Last published from https://github.com/zalmoxisus/devui
 - Changes since last publish:
   - Fix and update packages (https://github.com/reduxjs/redux-devtools/pull/449)
     - Lint changes
   - Move remotedev-app and remotedev-server (https://github.com/reduxjs/redux-devtools/pull/445)
     - Rename `remotedev` to `redux-devtools` for two constants
   - Add devui package (https://github.com/reduxjs/redux-devtools/pull/444) (reconstructed diff: https://github.com/Methuselah96/redux-devtools/commit/c2e5626db604e0018f48a517da707392f18b991b)
     - Lint changes
   - Upgrade to React 16 for storybook (https://github.com/zalmoxisus/devui/pull/21)
     - No change to src
     - Bump `color` dependency from `^1.0.1` to `^2.0.0`
     - Bump `prop-types` dependency from `^15.5.10` to `^15.6.0`
     - Bump `react-icons` dependency from `^2.2.5` to `^2.2.7`
     - Bump `react-jsonschema-form` dependency from `^0.48.2` to `^1.0.0`
     - Bump `react-select` dependency from `^1.0.0-rc.5` to `^1.0.0-rc.10`
     - Bump `styled-components` dependency from `^2.0.0` to `^2.2.2`
 - Changes that might have been missed from source repo:
   - None


#### redux-devtools-cli

 - Last published from this repo
 - Changes since last publish:
   - Fix and update packages (https://github.com/reduxjs/redux-devtools/pull/449)
     - Lint changes
   - Move remotedev-app and remotedev-server (https://github.com/reduxjs/redux-devtools/pull/445) (diff since publish: https://github.com/reduxjs/redux-devtools/pull/445/files/7ef6812ee46199a420f9e4615ffb1426d628258a..6de0b367fd16ad5434f3249bd69880edc16ad3ba)
     - Serve umd modules for cli (https://github.com/reduxjs/redux-devtools/pull/445/commits/8810fe42b6b65370663fcf78a569340a7df808ca)
     - Open Redux DevTools in browser from cli (https://github.com/reduxjs/redux-devtools/pull/445/commits/b00290387ec3c4d4120c1e47815ea501063fb61c)
     - Open as a standalone electron app (https://github.com/reduxjs/redux-devtools/pull/445/commits/22dcd97b9b532ee1240c678a75298d3b320717f6)
 - Changes that might have been missed from source repo:
   - None

#### d3tooltip

 - Last published from https://github.com/romseguy/d3tooltip
 - Changes since last publish:
   - Fix and update packages (https://github.com/reduxjs/redux-devtools/pull/449)
     - Lint changes
   - Merge map2tree package (https://github.com/reduxjs/redux-devtools/pull/426)
     - gitignore dist directory
   - Merge d3tooltip package (https://github.com/reduxjs/redux-devtools/pull/423) (reconstructed diff: https://github.com/Methuselah96/redux-devtools/commit/79749f898d9df7d314413b4a01bb0c71f52c380c)
     - No changes
   - Remove example (https://github.com/romseguy/d3tooltip/commit/55b594bdb1274ef110b7e76a7c7ef12a1cbf5e9f)
     - Does not affect src
 - Changes that might have been missed from source repo:
   - None

#### map2tree

 - Last published from https://github.com/romseguy/map2tree
 - Changes since last publish:
   - Fix and update packages (https://github.com/reduxjs/redux-devtools/pull/449)
     - Lint changes
   - Merge map2tree package (https://github.com/reduxjs/redux-devtools/pull/426) (reconstructed diff: https://github.com/Methuselah96/redux-devtools/commit/038011238253e13ddbc22f3c23f8ac4426012a7f)
     - No changes
 - Changes that might have been missed from source repo:
   - None

#### redux-devtools-test-generator

 - Last published from https://github.com/zalmoxisus/redux-devtools-test-generator
 - Changes since last publish:
   - Fix and update packages (https://github.com/reduxjs/redux-devtools/pull/449)
     - Lint changes
   - Move remotedev-app and remotedev-server (https://github.com/reduxjs/redux-devtools/pull/445)
     - Change devDependency `remotedev-inspector-monitor` to `redux-devtools-inspector`
   - Upgrade to React 16.4 for trace-monitor and test-generator (https://github.com/reduxjs/redux-devtools/pull/443)
     - Bump `react` dependency from `^15.1.0` to `^16.4.0`
   - Move redux-devtools-test-generator package (https://github.com/reduxjs/redux-devtools/pull/438) (reconstructed diff: https://github.com/Methuselah96/redux-devtools/commit/556fe2d3c1852302def21e8d40eb940f9b909b39)
     - Remove demo; no src changes
 - Changes that might have been missed from source repo:
   - None

#### redux-devtools-trace-monitor

 - Last published from this repo
 - Changes since last publish:
   - None
 - Changes that might have been missed from source repo:
   - None

**Comment by Methuselah96** [comment]
@timdorr After thinking about it a little more, I think it would make sense to just publish `react-json-tree` (in order to resolve https://github.com/reduxjs/redux-devtools/issues/480 and https://github.com/reduxjs/redux-devtools/issues/490) and `redux-devtools` (in order to resolve https://github.com/reduxjs/redux-devtools/pull/482).

`redux-devtools-inspector` can't be published until https://github.com/reduxjs/redux-devtools/issues/430 and https://github.com/reduxjs/redux-devtools/issues/431 are resolved.

Based on https://github.com/reduxjs/redux-devtools/issues/454#issuecomment-456973394, I don't think `redux-devtools-core` or `redux-devtools-cli` are finished yet, and so there's no need to publish those.

The rest of the packages don't have any non-internal changes.

**Comment by Methuselah96** [comment]
I don't know if it's helpful or not but here's a graph of the different repos getting merged together and the commit history since then: https://methuselah96.github.io/redux-devtools-app-testing/.

**Comment by stclairdaniel** [comment]
Any updates here? I'd love an update - we're getting warnings in our test runs about deprecated React methods.

**Comment by Methuselah96** [comment]
No updates from me. @timdorr Is there anything I can do to help publish? I have a bit of free time on my hand and it would help me motivate to work more on this project if I knew that fixes/changes are going to get published at some point.

**Comment by timdorr** [comment]
Hey @zalmoxisus or @alexkuz, can either of you grant me contributor access on npm? I'd like to help @Methuselah96 to get these updates published.

**Comment by Methuselah96** [comment]
@zalmoxisus @alexkuz @gaearon @jhen0409 @romseguy More specifically, the packages it would be great to have npm access to are:

Already in this monorepo/organization:
- `react-json-tree` (@alexkuz, @gaearon, @zalmoxisus)
- `redux-devtools-core` (@zalmoxisus)
- `redux-slider-monitor` (@jhen0409) (rename to `redux-devtools-slider-monitor`?)
- `redux-devtools-inspector` (@alexkuz, @zalmoxisus)
- `devui` (@zalmoxisus)
- `redux-devtools-cli` (@zalmoxisus)
- `d3tooltip` (@romseguy)
- `map2tree` (@romseguy)
- `d3-state-visualizer` (@romseguy)
- `redux-devtools-chart-monitor` (@romseguy)
- `redux-devtools-test-generator` (@zalmoxisus)
- `redux-devtools-trace-monitor` (@zalmoxisus)

Hope to move to this monorepo:
- `redux-devtools-extension` (@zalmoxisus)
- `react-base16-styling` (@alexkuz)
- `base16` (@gaearon)
- `redux-devtools-themes` (@gaearon)
- `remotedev-serialize` (@zalmoxisus) (we might rename this one anyway)

We already have access to `redux-devtools-instrument`, `redux-devtools`, and `redux-devtools-log-monitor`. You can give access to timdorr and/or methuselah96 on npm.

**Comment by zalmoxisus** [comment]
Hey guys! Thanks for all the effort here!
I granted access to @timdorr and @Methuselah96:
- `redux-slider-monitor`
- `d3-state-visualizer`
- `map2tree`
- `react-json-tree`
- `d3tooltip`
- `redux-devtools-chart-monitor`
- `redux-devtools-inspector`
- `redux-devtools-test-generator`
- `redux-devtools-extension`
- `remotedev-serialize`
- `devui`
- `redux-devtools-trace-monitor`
- `redux-devtools-core`
- `redux-devtools-cli`

Also for `redux-devtools-dock-monitor`, which is not used for the extension (and it's not in the list), but many are using it for "vanilla Redux Devtools".

Let me know if I missed some that I have access.

**Comment by Methuselah96** [comment]
Published: https://github.com/reduxjs/redux-devtools/releases/tag/react-json-tree%400.12.0

**Comment by alexkuz** [comment]
@timdorr @Methuselah96 here you are :) Thank you!
